docs/platforms/javascript/common/tracing/span-metrics/examples.mdx

```diff
@@ -19,86 +19,82 @@ This guide provides practical examples of using span attributes and metrics to s
 **Solution:** Track the entire file processing pipeline with detailed metrics at each stage, from client-side upload preparation through server-side processing.
 
 **Frontend Instrumentation:**
+
 ```javascript
 // Client-side file upload handling
 Sentry.startSpan(
   {
-    name: 'Client File Upload',
-    op: 'file.upload.client',
+    name: "Client File Upload",
+    op: "file.upload.client",
     attributes: {
       // Static details available at the start
-      'file.size_bytes': 15728640,  // 15MB
-      'file.type': 'image/jpeg',
-      'file.name': 'user-profile.jpg',
-      'client.compression_applied': true
-    }
+      "file.size_bytes": 15728640, // 15MB
+      "file.type": "image/jpeg",
+      "file.name": "user-profile.jpg",
+      "client.compression_applied": true,
+    },
   },
-  async () => {
-    // Get the current active span to update during upload
-    const span = Sentry.getActiveSpan();
-    
+  async (span) => {
     try {
       // Begin upload process
       const uploader = new FileUploader(file);
-      
+
       // Update progress as upload proceeds
-      uploader.on('progress', (progressEvent) => {
-        if (span) {
-          span.setAttribute('upload.percent_complete', progressEvent.percent);
-          span.setAttribute('upload.bytes_transferred', progressEvent.loaded);
-        }
+      uploader.on("progress", (progressEvent) => {
+        span.setAttribute("upload.percent_complete", progressEvent.percent);
+        span.setAttribute("upload.bytes_transferred", progressEvent.loaded);
       });
-      
-      uploader.on('retry', (retryCount) => {
-        if (span) {
-          span.setAttribute('upload.retry_count', retryCount);
-        }
+
+      uploader.on("retry", (retryCount) => {
+        span.setAttribute("upload.retry_count", retryCount);
       });
-      
+
       const result = await uploader.start();
-      
+
       // Set final attributes after completion
-      if (span) {
-        span.setAttribute('upload.total_time_ms', result.totalTime);
-        span.setAttribute('upload.success', true);
-        span.setAttribute('upload.server_file_id', result.fileId);
-      }
-      
+      span.setAttribute("upload.total_time_ms", result.totalTime);
+      span.setAttribute("upload.success", true);
+      span.setAttribute("upload.server_file_id", result.fileId);
+
       return result;
     } catch (error) {
       // Record failure information
-      if (span) {
-        span.setAttribute('upload.success', false);
-        span.setAttribute('upload.error_type', error.name);
-        span.setAttribute('upload.error_message', error.message);
-        span.setStatus({ code: 'ERROR' });
-      }
+      span.setAttribute("upload.success", false);
+      span.setAttribute("upload.error_type", error.name);
+      span.setAttribute("upload.error_message", error.message);
+
       throw error;
```

Comment:
m: I'd rather use our error monitoring product here instead of adding metrics. Maybe there's a case for `upload.error_type` if we want some kind of frequency/distribution gauge but given errors are trace-connected, capturing an error makes more sense to me.

Also, if we avoid rethrowing the error but capture here, we can associate the error with this very span and its scope data, rather than whatever outer scope is active where we'd catch it.

```suggestion
      span.setAttribute("upload.success", false);
      Sentry.captureException(error);
```

Reply:
re-throwing here should be fine and it should be correctly assigned to this span then anyhow. I agree on the error details being captured here, though, maybe we can just remove the whole `catch` block here 🤔

Reply:
Thats a fair point - im fine wiht that. My logic was that if someone was reviewing the span metrics directly as part of troubleshooting - having the visibility close to line of sight would be useful, but if we're capturing it as part of product they are going to see the error. Im fine with the change.